.github/workflows/community-label.yml

```diff
@@ -0,0 +1,112 @@
+name: Add Community Label
+
+on:
+  issues:
+    types: [opened]
+  pull_request_target:
+    types: [opened]
+
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  verify:
+    name: Verify
+    runs-on: ubuntu-20.04
+    outputs:
+      issueId: ${{ steps.check.outputs.issueId }}
+      run: ${{ steps.check.outputs.run }}
+    steps:
+      - name: Check if user is a community contributor
+        id: check
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const types = [ 'issue', 'pull_request' ];
+            const orgs = [ 'automattic', 'woocommerce' ];
+
+            let found = null; // null indicates we haven't done any checks.
+            let type, org, result;
+
+            for ( var typeIndex in types ) {
+              type = types[ typeIndex ];
+
+              if ( ! context.payload[ type ] || ! context.payload[ type ].user || ! context.payload[ type ].user.login ) {
+                // If we don't have the context for this type, move on to checking the next type.
+                console.log( 'No payload found for %s, continuing...', type );
+                continue;
+              }
+
+              if ( context.payload[ type ].user.type && context.payload[ type ].user.type === 'Bot' ) {
+                // Treat bots as non-community contributors.
+                console.log( 'User %s is a bot user. Skipping...', payload[ type ].user.login );
+                found = true;
+                break;
+              }
+
+              for ( var orgIndex in orgs ) {
+                org = orgs[ orgIndex ];
+
+                try {
+                  // First attempt to check memberships.
+                  result = await github.request( 'GET /orgs/{org}/memberships/{username}', {
+                    org,
+                    username: context.payload[ type ].user.login,
+                  } );
+                } catch ( error ) {
+                  if ( error.status === 404 ) {
+                    // This means the user wasn't found in the organization.
+                    found = false;
+                    continue;
+                  }
+
+                  // For any other error, we should also try checking public memberships.
+                  try {
+                    result = await github.request( 'GET /orgs/{org}/public_members/{username}', {
+                      org,
+                      username: context.payload[ type ].user.login,
+                    } );
+                  } catch ( error ) {
+                    if ( error.status === 404 ) {
+                      // This means the user wasn't found in the organization.
+                      found = false;
+                      continue;
```

Comment:
many breaks and continues are really not idiomatic JS either, I think this is a symptom of the fact you're trying to maintain state in let variables instead of using JavaScripts many collection filtering/manipulation options available. In my opinion it is very hard to follow when doing follow up maintenance work on this kind of code.